lib/pf/role/pool.pm

```diff
@@ -0,0 +1,179 @@
+package pf::role::pool;
+
+=head1 NAME
+
+pf::role::pool - Object oriented module for VLAN pool
+
+=head1 SYNOPSIS
+
+The pf::role::pool module contains the functions necessary for the VLAN Pool.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+
+use pf::config;
+use pf::util qw (%Config);
+use pf::log();
+
+use pf::node;
+
+use Number::Range;
+
+=head1 SUBROUTINES
+
+Warning: The list of subroutine is incomplete
+
+=cut
+
+=head2 new
+
+Constructor.
+
+=cut
+
+sub new {
+    my $logger =  pf::log::get_logger();
+    $logger->debug("instantiating new pf::role::pool object");
+    my ( $class, %argv ) = @_;
+    my $this = bless {}, $class;
+    return $this;
+}
+
+sub getVlanFromPool {
+    my ($self, $args) = @_;
+    my $logger =  pf::log::get_logger();
+
+    return $pool_args->{'vlan'} if $self->rangeValidator($args->{'vlan'});
+    my $range = Number::Range->new($args->{'vlan'});
+    my $vlan;
+    if ($Config{advanced}{vlan_pool_technique} eq "username_hash") {
```

Comment:
constant for "username_hash"

Reply:
Also maybe just a debug or trace statement to say which technique it will be using